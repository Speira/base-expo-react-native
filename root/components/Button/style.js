import styled from 'styled-components/native'
import { TouchableOpacity, Text, View } from 'react-native'

/**
 * CardHeader
 * @component
 *
 */
export const ButtonText = styled(Text)`
  color: ${({ theme }) => theme.COLORS.STATIC.LIGHT};
`

/**
 * ButtonWiew
 * @component
 *
 */
export const ButtonWiew = styled(View)`
  justify-content: center;
  padding: 10px;
`
/**
 * BaseButton
 * @component
 *
 */
const BaseButton = styled(TouchableOpacity)`
  align-items: center;
  background-color: ${({ theme, status }) => theme.COLORS.DYNAMIC[status]};
  padding: 10px;
`
export default BaseButton
