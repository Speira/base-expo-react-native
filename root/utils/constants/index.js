// utils/constants
// description: We try to set all constants into a big one to reduce Javascript
//              constants performance issue.

export default {
  API: {
    FETCH_DATA_URL:
      'https://static.leboncoin.fr/img/shared/technical-test.json',
    PROXY: {
      NO_CORS_SERVER: 'https://cors-anywhere.herokuapp.com/',
    },
    TIMEOUT: 20000,
  },
  LABELS: {
    CLEAR_DATABASE: 'Clear the database',
    START_DATABASE: 'Create the database',
    IMVALID_DATA_FORMAT: ' Invalid data format',
    LIST: 'Item List',
    NO_CARD: 'No card to persist',
    NO_DATABASE: 'The database is not set',
    NO_RESPONSE_DATA: 'There is no data in the query response',
    REFETCH_DATA: 'Refetch Data',
    SEARCH_DATA_FAILED:
      'No data for SearchDataAPI: Please unlock the proxy below and then refetch data',
    WELCOME: 'Welcome to the ListItem app',
  },
  STATES: {
    ERROR: 'error',
    LOADING: 'loading',
  },
  STATUS: {
    DANGER: 'DANGER',
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
  STORAGE_ITEMS: {
    DB: 'storage_list_item_app',
    CARD: 'card_items.db',
    THEME: 'app_theme_storage',
  },
  THEMES: {
    DEFAULT: {
      COLORS: {
        STATIC: {
          PRIMARY: '#EBD8C3',
          SECONDARY: '#F7E9D7',
          TERTIARY: '#FFF6EA',
          QUATERNARY: '#FFEEEE',
          LIGHT: '#f8f8fa',
          DARK: '#4d3e30',
        },
        DYNAMIC: {
          INFO: '#035397',
          SUCCESS: '#1C7947',
          WARNING: '#C36A2D',
          DANGER: '#A9333A',
        },
      },
    },
  },
}
