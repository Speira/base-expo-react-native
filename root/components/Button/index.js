import PropTypes from 'prop-types'
import { TouchableOpacity, Text } from 'react-native'
import constants from '~utils/constants'
import BaseButton, { ButtonText, ButtonWiew } from './style'

const { STATUS } = constants

/**
 * Button
 * @component
 *
 */
function Button(props) {
  const { hide, label, onPress, status } = props
  return hide ? null : (
    <ButtonWiew>
      <BaseButton onPress={onPress} status={status}>
        <ButtonText>{label}</ButtonText>
      </BaseButton>
    </ButtonWiew>
  )
}
Button.defaultProps = {
  hide: false,
  onPress: () => null,
  status: STATUS.INFO,
}
Button.propTypes = {
  hide: PropTypes.bool,
  onPress: PropTypes.func,
  label: PropTypes.string.isRequired,
  status: PropTypes.oneOf(Object.values(STATUS)),
}

export default Button
