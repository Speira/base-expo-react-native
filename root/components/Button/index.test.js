import React from 'react'
import { shallow } from 'enzyme'
import constants from '~utils/constants'
import Button from './index'

const { THEMES } = constants

describe('render Button', () => {
  const wrapper = shallow(<Button label="test" theme={THEMES.DEFAULT} />)
  it('Button should be defined', () => {
    expect(wrapper).toBeDefined()
  })
  it('Button should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
