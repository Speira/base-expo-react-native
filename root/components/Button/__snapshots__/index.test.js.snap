// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button Button should match snapshot 1`] = `
<style__ButtonWiew>
  <style__BaseButton
    onPress={[Function]}
    status="INFO"
  >
    <style__ButtonText>
      test
    </style__ButtonText>
  </style__BaseButton>
</style__ButtonWiew>
`;
